src/features/Instructors/data/thunks.js

```diff
@@ -63,18 +61,29 @@ function assignInstructors(data) {
 /**
  * Add instructor to an institution.
  * @param {number} institutionId - The ID of the institution to which it will be assigned.
- * @param {string} instructorEmail - The instructor email to be added.
+ * @param {FormData} instructorFormData - Instructor formData {instructorEmail, firstName, lastName}.
  * @returns {Promise<void>} - A promise that resolves after dispatching appropriate actions.
  */
-function addInstructor(institutionId, instructorEmail) {
+function addInstructor(institutionId, instructorFormData) {
   return async (dispatch) => {
-    dispatch(addInstructorRequest());
+    dispatch(updateInstructorAdditionRequest({ status: RequestStatus.LOADING }));
     try {
-      const response = await handleNewInstructor(institutionId, instructorEmail);
-      dispatch(addInstructorSuccess(response.data));
+      const { data } = await handleNewInstructor(institutionId, instructorFormData);
+      dispatch(updateInstructorAdditionRequest({ status: RequestStatus.SUCCESS, data }));
     } catch (error) {
-      dispatch(addInstructorFailed());
-      logError(error);
+      let errors = '';
+      const { customAttributes } = error || {};
+      const { httpErrorResponseData, httpErrorStatus } = customAttributes || {};
+
+      if (httpErrorStatus === 500) {
+        errors = 'Whoa! Something went wrong, maybe the user already exist, or our server just take a break.';
```

Comment:
This message is too familiar with the user, what should be a correct message for this case @anfbermudezme ?